eslint-plugin/rules/validate-imports.js

```diff
@@ -174,28 +174,25 @@ module.exports = {
           IMPORTED_FRAGMENTS = Object.create(null);
           IMPORT_LINE_USED = new Map();
 
-          const importLinesToInlinedSource = inlineImports.lineToImports(
-            context.getSourceCode().text,
-            {
-              resolveOptions: {
-                basedir: path.dirname(filename),
-              },
-              resolveImport,
-              throwIfImportNotFound: false,
-            },
-          );
+          const lineToFragmentNameToBucket = gatherFragmentImports({
+            source: context.getSourceCode().text,
+            sourceLocation: filename,
+            resolveImport,
+            fragmentParserGenerator: context.parserServices.getFragmentDefinitionsFromSource,
+            throwIfImportNotFound: false,
+          });
 
-          for (const [lineNumber, source] of importLinesToInlinedSource) {
+          for (const lineNumber of Object.keys(lineToFragmentNameToBucket)) {
```

Comment:
you're reading the value anyway so you might as well iterate over `Object.entries`